kernel/kernel-api/src/test/scala/io/delta/kernel/utils/CloseableIteratorSuite.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.utils
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.must.Matchers.contain
+import org.scalatest.matchers.should.Matchers.convertToAnyShouldWrapper
+
+import java.util
+
+class CloseableIteratorSuite extends AnyFunSuite {
+
+  test("test take while with all elements matching filter") {
```

Comment:
`test("test ` ---> don't need to say "test" at the start of a test name

`take while` --> `takeWhile`